crates/lib/src/diagnostics.rs

```diff
@@ -0,0 +1,212 @@
+//! Produce diagnostic information about the crate and its runtime environment in order to aid
+//! in debugging and remote user support.
+
+use std::{borrow::Cow, time::Duration};
+
+use qcs_api_client_openapi::models::User;
+
+use crate::{build_info, client::Qcs, qvm::QvmOptions};
+
+/// Collect package diagnostics in string form
+pub async fn gather() -> String {
+    Diagnostics::gather().await.to_string()
+}
+
+/// Diagnostic information representing the environment in which this crate
+/// was built and is executed, for use in diagnosing unexpected and incorrect
+/// behavior.
+#[derive(Debug)]
+struct Diagnostics {
+    /// The version of this crate    
+    version: String,
+
+    rust_version: String,
+
+    /// The features with which this crate was compiled
+    features: Vec<&'static str>,
+
+    qcs: QcsApiDiagnostics,
+    quilc: QuilcDiagnostics,
+    qvm: QvmDiagnostics,
+}
+
+impl Diagnostics {
+    async fn gather() -> Self {
+        let client = Qcs::load().await;
+
+        let (qcs, qvm) = futures::future::join(
+            QcsApiDiagnostics::gather(&client),
+            QvmDiagnostics::gather(&client),
+        )
+        .await;
+
+        Self {
+            version: build_info::PKG_VERSION.to_owned(),
+            rust_version: build_info::RUSTC_VERSION.to_owned(),
+            features: build_info::FEATURES
+                .iter()
+                .map(std::borrow::ToOwned::to_owned)
```

Comment:
Could you use `copied()` or `cloned()` instead of this `map()`?

Reply:
yeah, not sure what I was thinking here

Reply:
`to_vec()` works in this case, so I went with that.